frontend/middleware_validatestatic.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+// Referenced in https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftresources
+var rxResourceGroupName = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+
+func MiddlewareValidateStatic(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+
+	subId := chi.URLParam(r, "subscriptionId")
+	resourceGroupName := chi.URLParam(r, "resourceGroupName")
+	resourceProviderNamespace := chi.URLParam(r, "resourceProviderNamespace")
+	resourceType := chi.URLParam(r, "resourceType")
+	operationId := chi.URLParam(r, "operationId")
+	resourceName := chi.URLParam(r, "resourceName")
+
+	apiVersion := r.URL.Query().Get(APIVersionKey)
+
+	if r.URL.Path != strings.ToLower(r.URL.Path) {
+		if log, ok := r.Context().Value(contextKeyOriginalPath).(logr.Logger); ok {
+			log.Error(errors.New("LowerCase error"), "path was not lower case")
+		}
+		arm.WriteError(w, http.StatusInternalServerError, arm.CloudErrorCodeInternalServerError, "", "Internal server error.")
```

Comment:
There's a helper function for this specific error since it's so frequest: `arm.WriteInternalServerError`